src/android/android-backend.cpp

```diff
@@ -0,0 +1,73 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2015 Intel Corporation. All Rights Reserved.
+#ifdef RS2_USE_ANDROID_BACKEND
```

Comment:
This ifdef is repeated throughout in various places (although not all). Is it necessary? These files so far as I can tell are only built by cmake when we are building with the Android NDK toolchain which will set the `RS2_USE_ANDROID_BACKEND` . Its presence kind of implies we could end up in a situation where this is built without this being set, but I can't see how that's possible. Have I missed something here?

Reply:
No, it's not necessary, and I dislike it too, but I left it in because we have similar macros for the Windows and V4L backends.

I think an `#if __ANDROID__` check would be better, but fixing this wasn't part of the goal here :)

Reply:
Reasonable - if you wanted to remove this to set an example in the style of "no broken windows" I would support it but I don't mind if you want to keep it.

Reply:
Considering how much time everyone's spent around this set of changes, I'd rather keep it as it is for now.